lib/ohloh_scm/adapters/git/misc.rb

```diff
@@ -76,14 +76,32 @@ def no_tags?
 
     def tags
       return [] if no_tags?
-      tag_strings = run("cd #{url} && git tag --format='%(creatordate:iso-strict) %(objectname) %(refname)'").split(/\n/)
+      tag_strings = run("cd #{url} && git tag --format='%(creatordate:iso-strict) %(objectname) %(refname)' | sed 's/refs\\/tags\\///'").split(/\n/)
       tag_strings.map do |tag_string|
-        timestamp_string, commit_hash, tag_path = tag_string.split(/\s/)
-        timestamp_string = '1970-01-01' if timestamp_string.strip.empty?
-        timestamp = Time.parse(timestamp_string)
-        tag_name = tag_path.gsub('refs/tags/', '')
-        [tag_name, commit_hash, timestamp]
+        timestamp_string, commit_hash, tag_name = tag_string.split(/\s/)
+        [tag_name, dereferenced_sha(tag_name) || commit_hash, time_object(timestamp_string)]
       end
     end
-	end
+
+    private
+
+    def dereferenced_sha(tag_name)
+      dtag_sha_and_name = dtag_sha_and_names.find { |sha_and_name| sha_and_name.last == tag_name }
+      dtag_sha_and_name.first if dtag_sha_and_name
+    end
+
+    def dtag_sha_and_names
+      @dtag_sha_and_names ||= derefenced_tag_strings.map { |string| string.split(/\s/) }
```

Comment:
This part can also be written as
```ruby
      dereferenced_tag = dereferenced_tags.find { |dtag| dtag[1] == tag_name }
      dereferenced_tag[0] if dereferenced_tag
    end

    def dereferenced_tags
      @dereferenced_tags ||= derefenced_tag_strings.map { |string| string.split(/\s/) }
```
I went with the former because I felt that `dereferenced_tags` was too ambiguous, especially when one does `dereferenced_tag[0]`.